spec.bs

```diff
@@ -652,10 +653,51 @@ To <dfn>generate and score bids</dfn> given an [=auction config=] |auctionConfig
 To <dfn>score and rank a bid</dfn> given an [=auction config=] |auctionConfig|, a [=generated bid=]
 |generatedBid|, a [=leading bid info=] |leadingBidInfo|, a [=string=] |decisionLogicScript|, and a
 [=boolean=] |hasComponentAuction|:
-1. TODO: Assemble and fetch trusted scoring signals
-1. TODO: Let |adMetadata| be ... maybe include in [=generated bid=]
+1. Let |renderUrls| be a new [=list/is empty|empty=] [=list=].
+1. Let |adComponentRenderUrls| be a new [=list/is empty|empty=] [=list=].
+1. If |generatedBid|'s [=generated bid/ad descriptor=] is not null:
+  1. [=list/Append=] |generatedBid|'s [=generated bid/ad descriptor=]'s [=ad descriptor/url=] to
+    |renderUrls|.
+1. If |generatedBid|'s [=generated bid/ad component descriptors=] is not null:
+  1. [=list/For each=] |adComponentDescriptor| in |generatedBid|'s
+    [=generated bid/ad component descriptors=]:
+    1. [=list/Append=] |adComponentDescriptor|'s [=ad descriptor/url=] to |adComponentRenderUrls|.
+1. Let |fullSignalsUrl| be the result of [=building trusted scoring signals url=] with |auctionConfig|'s
+  [=auction config/trusted scoring signals url=], |renderUrls|, |adComponentRenderUrls|,
+  |auctionConfig|'s [=auction config/seller experiment group id=].
+
+  Note: Implementations can batch requests by collecting render URLs and ad component render URLs
+  from multiple invocations of [=score and rank a bid=] and passing them all to a single invocation
+  of [=building trusted scoring signals url=] -- the network response has to be parsed to pull out the pieces
+  relevant to each scoreAd() call.
+1. Let |trustedScoringSignalsBytes| be the result of [=fetching resource=] with |fullSignalsUrl|
+  and "application/json". TODO: share some logic here with trustedBiddingSignals.
+1. TODO: If |trustedScoringSignalsBytes| is failure, fail the auction with manually_aborted set to
+  true.
+1. Let |trustedScoringSignalsParsed| be the result of [=parsing JSON bytes to an Infra value=] with
+  |trustedScoringSignalsBytes|.
+1. TODO: If |trustedScoringSignalsParsed| is failure, or if or if |trustedScoringSignalsParsed| is not an

```

Comment:
Same as the other comment, we're not failing the auction when [parsing](https://source.chromium.org/chromium/chromium/src/+/refs/heads/main:content/services/auction_worklet/trusted_signals.cc;drc=a61b95c63b0b75c1cfe872d9c8cdf927c226046e;l=517) trusted signals failed.

Reply:
Done, but see my other comment.